# One-Sample z-test - Lab

## Introduction
In this lab you'll perform a few quick tests to help you better understand how hypothesis testing works.

## Objectives
You will be able to:
* Understand and explain use cases for a one-sample z-test
* Set up null and alternative hypotheses
* Calculate z-statistic using z-tables and CDF functions
* Calculate and interpret p-value for significance of results

## Exercise 1
A fast food chain claims that the mean time to order food at their restaurants is 60 seconds, with a standard deviation of 30 seconds. You decide to take this claim to the test and go to one of the restaurants to observe actual waiting times. You take a sample of 36 customers and find that the mean order time was 75 seconds. Does this finding prove enough evidence to contradict the fast food chain's claim of fast service?

Follow the 5 steps shown in previous lesson and use $\alpha$ = 0.05. 

#### State your null and alternative hypotheses:

Does ordering at this fast food chain take longer than they claim?

Alternative Hypothosis ($H_a$): average ordering time ($\mu$) < observed ordering time ($M$)

Null Hypothosis ($H_0$): average ordering time ($\mu$) $\geq$ observed ordering time ($M$)

In [1]:
# Imports

import scipy.stats as stats
from math import sqrt

In [2]:
# Setting variables

x_bar = 75   # Sample mean, 75 seconds
n = 36       # Sample size, number of customer orders observed
sigma = 30   # Population standard deviation, 30 seconds
mu = 60      # Population mean, 60 seconds

In [6]:
# Calculating a one-sample z-statistic (z-score)
z = (x_bar - mu)/(sigma/sqrt(n))
print(z)

# Using scipy.stats to calculating z-value
z_val = stats.norm.cdf(z)
print(z_val)

# The inverse of the z-value is the p-value, so calculating p-value
p_val = 1 - z_val
print(p_val)

# (p = 0.0013498980316301035, z = 3.0)

3.0
0.9986501019683699
0.0013498980316301035


#### Interpret the results in terms of the p-value

If we assume that the mean of ordering times is 60 seconds, our observed mean of 75 seconds is higher than 99.865% of the values of all ordering times.

Assuming an alpha threshold of .05, our results are statistically significant because the p-value we calculated is less than .05. Thus we can reject the null hypothosis, and we can say that statistically it is likely the mean of ordering times is greater than 60 seconds.

## Exercise 2

25 students complete a preparation program for taking the SAT test.  Here are the SAT scores from the 25 students who completed  program:

``
434 694 457 534 720 400 484 478 610 641 425 636 454 
514 563 370 499 640 501 625 612 471 598 509 531
``

We know that the population average for SAT scores is 500 with a standard deviation of 100.

Are our 25 students’ SAT scores significantly bigger than a population mean? 

*Note that the SAT preparation program claims that it will increase (and not decrease) the SAT score.  So, you can conduct a one-directional test. (alpha = .05).*

#### State your null and alternative hypotheses:

Alternative Hypothosis ($H_a$): average SAT scores ($\mu$) < prep program students' SAT scores ($M$)

Null Hypothosis ($H_0$): average SAT test scores ($\mu$) $\geq$ prep program students' SAT scores ($M$)

In [9]:
# Imports
import scipy.stats as stats
from math import sqrt
import numpy as np

# Calculating the average from the provided SAT scores
prep_scores = [434, 694, 457, 534, 720, 400, 484, 478, 610, 641, 425, 636, 454, 
               514, 563, 370, 499, 640, 501, 625, 612, 471, 598, 509, 531]

# Setting variables
x_bar = np.mean(prep_scores) # Sample mean, from our prep program SAT scores
n = 25                       # Sample size, number of students' SAT scores
sigma = 100                  # Population standard deviation, 100 
mu = 500                     # Population mean, 500


print(x_bar)

536.0


In [8]:
# Calculating a one-sample z-statistic (z-score)
z = (x_bar - mu)/(sigma/sqrt(n))
print(z)

# Using scipy.stats to calculating z-value
z_val = stats.norm.cdf(z)
print(z_val)

# The inverse of the z-value is the p-value, so calculating p-value
p_val = 1 - z_val
print(p_val)

# p = 0.03593031911292577, z = 1.8

1.8
0.9640696808870742
0.03593031911292577


#### Interpret the results in terms of the p-value

Assuming a threshold alpha of .05, our results are statistically significant. Given the SAT scores of our prep program students, their scores show a statistically significant increase in their SAT scores as a result of taking the prep program.

## Summary

In this lesson, you conducted a couple of simple tests comparing sample and population means, in an attempt to reject our null hypotheses. This provides you with a strong foundation to move ahead with more advanced tests and approaches later on.